Source/AviationWeather.Test/Source/metar_parser_tests.cpp

```diff
@@ -0,0 +1,575 @@
+    Assert::AreEqual(uint32_t(300), metar.sky_condition_group.at(0).layer_height);
+    Assert::AreEqual(distance_unit::feet, metar.sky_condition_group.at(0).unit);
+    metar.sky_condition_group.clear();
+
+    // BKN060CB
+    std::string m4("BKN060CB ");
+    metar::parsers::parse_sky_condition(metar, m4);
+    Assert::AreEqual(sky_cover_type::broken, metar.sky_condition_group.at(0).sky_cover);
+    Assert::AreEqual(sky_cover_cloud_type::cumulonimbus, metar.sky_condition_group.at(0).cloud_type);
+    Assert::AreEqual(uint32_t(6000), metar.sky_condition_group.at(0).layer_height);
+    Assert::AreEqual(distance_unit::feet, metar.sky_condition_group.at(0).unit);
+    metar.sky_condition_group.clear();
+
+    // FEW008 SCT030
+    std::string m5("FEW008 SCT030 ");
+    metar::parsers::parse_sky_condition(metar, m5);
+    Assert::AreEqual(sky_cover_type::few, metar.sky_condition_group.at(0).sky_cover);
+    Assert::AreEqual(sky_cover_cloud_type::unspecified, metar.sky_condition_group.at(0).cloud_type);
+    Assert::AreEqual(uint32_t(800), metar.sky_condition_group.at(0).layer_height);
+    Assert::AreEqual(distance_unit::feet, metar.sky_condition_group.at(0).unit);
+    Assert::AreEqual(sky_cover_type::scattered, metar.sky_condition_group.at(1).sky_cover);
+    Assert::AreEqual(sky_cover_cloud_type::unspecified, metar.sky_condition_group.at(1).cloud_type);
+    Assert::AreEqual(uint32_t(3000), metar.sky_condition_group.at(1).layer_height);
+    Assert::AreEqual(distance_unit::feet, metar.sky_condition_group.at(1).unit);
+    metar.sky_condition_group.clear();
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_TemperatureDewpoint()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(int8_t(INT8_MAX), metar.temperature);
+    Assert::AreEqual(int8_t(INT8_MAX), metar.dewpoint);
+
+    // Temperature: 19C, Dewpoint: 4C
+    std::string m1("19/04 ");
+    metar::parsers::parse_temperature_dewpoint(metar, m1);
+    Assert::AreEqual(int8_t(19), metar.temperature);
+    Assert::AreEqual(int8_t(4), metar.dewpoint);
+
+    // Temperature: -3C, Dewpoint: -9C
+    std::string m2("M03/M09 ");
+    metar::parsers::parse_temperature_dewpoint(metar, m2);
+    Assert::AreEqual(int8_t(-3), metar.temperature);
+    Assert::AreEqual(int8_t(-9), metar.dewpoint);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_Altimeter()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(0.0, metar.altimeter_group.pressure);
+    Assert::AreEqual(pressure_unit::hPa, metar.altimeter_group.unit);
+
+    // QNH 1013
+    std::string m1("Q1013");
+    metar::parsers::parse_altimeter(metar, m1);
+    Assert::AreEqual(1013.0, metar.altimeter_group.pressure);
+    Assert::AreEqual(pressure_unit::hPa, metar.altimeter_group.unit);
+
+    // Altimeter 30.29
```

Comment:
Just for consistency with the test in millibars, can we use 29.92 inHg? (That way both represent standard pressure.)